src/core/Akka.Streams/Implementation/Fusing/ActorGraphInterpreter.cs

```diff
@@ -98,11 +133,63 @@ public GraphModule(GraphAssembly assembly, Shape shape, Attributes attributes, I
     [InternalApi]
     public sealed class GraphInterpreterShell
     {
+        internal sealed class BoxedRequestMore : ActorGraphInterpreter.IBoundaryEvent
+        {
+            public ActorGraphInterpreter.RequestMore Boxed { get; private set; }
+
+            public BoxedRequestMore()
+            {
+                
+            }
+            public BoxedRequestMore SetBox(ActorGraphInterpreter.RequestMore value)
+            {
+                Boxed = value;
+                return this;
+            }
+
+            public GraphInterpreterShell Shell => Boxed.Shell;
+        }
+        internal sealed class BoxedOnNext : ActorGraphInterpreter.IBoundaryEvent
```

Comment:
@to11mtm can you describe the approach / "why do it this way?" for me just so I understand the thinking behind this.